server/src/main/java/io/druid/segment/loading/SegmentLoaderLocalCacheManager.java

```diff
@@ -125,39 +145,58 @@ public File getSegmentFiles(DataSegment segment) throws SegmentLoadingException
       // We use a marker to prevent the case where a segment is downloaded, but before the download completes,
       // the parent directories of the segment are removed
       final File downloadStartMarker = new File(storageDir, "downloadStartMarker");
-      synchronized (lock) {
-        if (!storageDir.mkdirs()) {
-          log.debug("Unable to make parent file[%s]", storageDir);
-        }
-        try {
-          if (!downloadStartMarker.createNewFile()) {
-            throw new SegmentLoadingException("Was not able to create new download marker for [%s]", storageDir);
+      try {
```

Comment:
Could you please extract the body of this `try {}` block as a method? I know that the project is full of very long methods, but at least we can avoid adding them in the new or refactored code.